internal/transport/http2_client.go

```diff
@@ -866,21 +870,21 @@ func (t *http2Client) Close() error {
 	t.mu.Unlock()
 	t.controlBuf.finish()
 	t.cancel()
-	err := t.conn.Close()
+	closeErr := t.conn.Close()
```

Comment:
AFAICT nobody uses this error at all.  So optional: remove `error` as a return value from this function (and in the `ClientTransport` interface).

Reply:
Good point, done.